test/org/javarosa/xform/parse/FormParserHelper.java

```diff
@@ -14,21 +15,17 @@ private FormParserHelper() {
     }
 
     public static ParseResult parse(Path formName) throws IOException {
-        XFormParser parser = new XFormParser(new FileReader(formName.toString()));
+        return parseWithPrefix(formName, "");
+    }
+
+    public static ParseResult parseWithPrefix(Path formName, String externalInstancePathPrefix) throws IOException {
```

Comment:
Two comments on this one:
- Why not call it `parse`?
- I would argue that it's easier to have a second argument `externalInstances` of type `Path`
  - This would let the `parse(Path)` to be implemented as:

    ```java
    public static ParseResult parse(Path formName) throws IOException {
      return parse(formName, formName.getParent());
    }
    ```
  - This would let you avoid some of the changes in `XFormParserTest.java`

Reply:
Hey, thanks for all the great feedback. To overload parse would mean I could no longer use static imports of parse and would have to change all the callers. I was trying to avoid that.

Reply:
Oh, I see I can leave that static import alone and call the new two-args `parse` via `FormParserHelper.parse`.